client/src/pages/Homepage/SearchBox.jsx

```diff
@@ -224,249 +301,325 @@ const InputWrapper = styled.div`
+        if (block && parcels) {
+            // history.push(`/plans?block=${block}&parcel=${parcels}`);
+            // getting the block and parcel lat lon
+            const currentParcel = parcelList.find(
+                (parcel) => parcel.label === parcels
+            );
+            const [lng, lat] = JSON.parse(currentParcel.centroid)?.coordinates;
+            if (lat && lng) window.location.href = `/plans?loc=${lat},${lng}`;
+        } else if (!block && parcels) {
+            setBlockinputerror(true);
+        } else if (block && !parcels) {
+            setParcelinputerror(true);
+        } else {
+            setParcelinputerror(true);
+            setBlockinputerror(true);
+        }
+    }
+
+    async function handleInputChangeBlock(text) {
+        if (text) {
+            // setBlock(text);
+            setloadingAutocompleteBlock(true);
+            setBlockinputerror(false);
+
+            api.get(`/topfive?blockNum=${text}`)
+                .then((result) => {
+                    console.log({ result });
+                    if (result.status === 'failed') {
+                        throw new Error('No parcels found');
+                    }
+                    setblockList(result.data);
+                    setloadingAutocompleteBlock(false);
+                })
+                .catch((error) => {
+                    setBlockinputerror(true);
+                    setloadingAutocompleteBlock(false);
+                });
+        } else {
+            setblockList([]);
+            setloadingAutocompleteBlock(false);
+        }
+    }
+
+    function handleSubmitBlockDetails(blockNum) {
+        setBlock(blockNum);
+        dispatch(setBlockNum({ block: blockNum }));
+        setIsDisable(false);
+    }
+
+    function handleSubmitParcelDetails(parcelNum) {
+        setParcels(parcelNum);
+        dispatch(setParcel({ parcel: parcelNum }));
+    }
+
+    async function handleInputChangeParcel(text) {
+        if (text) {
+            // setParcels(text);
+            setParcelinputerror(false);
+            setloadingAutocompleteParcel(true);
+
+            api.get(`/topfive?blockNum=${block}&parcelNum=${text}`)
+                .then((result) => {
+                    if (result.status === 'failed') {
+                        throw new Error('No parcels found');
+                    }
+                    setparcelList(result.data);
+                    setloadingAutocompleteParcel(false);
+                })
+                .catch((error) => {
+                    console.log({ error });
+                    setloadingAutocompleteParcel(true);
+                    setParcelinputerror(true);
+                });
+        } else {
+            setloadingAutocompleteParcel(false);
+            setparcelList([]);
+        }
+    }
+
+    useEffect(() => {
+        locationAutocompleteApi.init();
+    }, []);
+
+    const [dropdownIsOpen, setDropdownIsOpen] = useState(false);
+
+    return (
```

Comment:
Why on clicking search (block/ parcel or address ) the page is refreshed? The search should be performed without a refresh

Reply:
@GalGend fixed, someone used window.location.href.